include/sys/zfs_context.h

```diff
@@ -273,6 +273,7 @@ typedef struct kmutex {
 } kmutex_t;
 
 #define	MUTEX_DEFAULT	0
+#define	MUTEX_FSTRANS	MUTEX_DEFAULT
```

Comment:
I'll try to save a bit of the precious behlendorf's time by answering this question:
Look at lines 34,75 and 742/743 of this file.
Most of this file (between lines 75 and 742) gets compiled only if <code>\_\_KERNEL\_\_</code> is not defined, so: in userspace.
This is useful for ztest and userspace utilities.
In userspace we don't need FSTRANS mutex-es, so it's simply redefined to a normal mutex.
Instead, in kernelspace, the definition that will be used by the compiler will be the one at https://github.com/behlendorf/spl/commit/b0d285706ca2ca5ae29791e2cf06a6a80274ae7d#diff-120f4508a55b150b43c2781b1ccbe485R36 , which gets included by sys/t_lock.h at line 38 of this file.
Is this clear enough?